libs/components/modals/src/lib/modules/modal/modal-adapter.service.ts

```diff
@@ -47,4 +48,50 @@ export class SkyModalAdapterService {
   #removeClassFromBody(className: string): void {
     this.#bodyEl.classList.remove(className);
   }
+
+  public hideHostSiblings(hostElRef: ElementRef): void {
+    const hostElement = hostElRef.nativeElement;
+    const hostSiblings = hostElement.parentElement.children;
+
+    for (let i = 0; i < hostSiblings.length; i++) {
+      const element = hostSiblings[i];
+      if (
+        element !== hostElement &&
+        !element.hasAttribute('aria-live') &&
+        element.nodeName.toLowerCase() !== 'script' &&
+        element.nodeName.toLowerCase() !== 'style'
+      ) {
+        // preserve previous aria-hidden status of elements outside of modal host
+        this.#hostSiblingPreviousValue.set(
+          element,
+          element.getAttribute('aria-hidden')
+        );
+        element.setAttribute('aria-hidden', 'true');
+      }
+    }
+  }
+
+  public unhideOrRestoreHostSiblings(): void {
+    this.#hostSiblingPreviousValue.forEach((previousValue, element) => {
+      // if element had aria-hidden status prior, restore status
+      if (previousValue) {
+        element.setAttribute('aria-hidden', previousValue);
+      } else {
+        element.removeAttribute('aria-hidden');
+      }
+    });
+    this.#hostSiblingPreviousValue.clear();
+  }
+
+  public hidePreviousModal(topModal: Element): void {
+    if (topModal && topModal.previousElementSibling) {
+      topModal.previousElementSibling.setAttribute('aria-hidden', 'true');
+    }
+  }
+
+  public unhidePreviousModal(topModal: Element): void {
```

Comment:
Adapter services should never receive an `Element` but rather an `ElementRef`. The idea behind this decision is that these adapter services can be switched out with other platform adapters (besides the DOM). For example, if we wanted to use the modal stuff in a mobile app, we could switch the DOM adapter with a mobile phone adapter. Since the mobile phone adapter wouldn't know what an `Element` is, we should always pass the `ElementRef` since that will represent either a DOM element or a mobile phone "element".

This should be updated in a few places above, too.